appeals/web/src/server/lib/mappers/data/initialise-and-map-data.js

```diff
@@ -0,0 +1,28 @@
+/** @type {import('./types.js').InitialiseAndMapDataFactory} */
+export const initialiseAndMapDataFactory = (
+	getSubmapperParams,
+	submaps,
+	getAppealType,
+	targetKey
+) => {
+	return async (innerParams, filterKeys) => {
```

Comment:
This is really nice in terms of the end result, but I worry about adding more layers of abstraction to our already multi-layered mapping system. I'm familiar with the codebase but I'm still struggling to wrap my head around all the moving parts in this function - do you think there's a way we could achieve the same result with simpler code that's easier to understand?

Reply:
I don't believe you could do this without a higher order function